src/depWalker.js

```diff
@@ -331,9 +332,11 @@ async function depWalker(manifest, options = Object.create(null)) {
         regSpinner.succeed(white().bold(i18n.getToken("depWalker.success_registry_metadata")));
     }
 
-    // Search for vulnerabilities in the local .json db
+    // Search for vulnerabilities relatively to the current initialized strategy
+    const { hydrateNodeSecurePayload } = await getVulnerabilityStrategy();
     await hydrateNodeSecurePayload(payload.dependencies);
 
```

Comment:
Here we also need to assign the vulnerability type to the payload (so i guess destructuring hydrateNodeSecurePayload is not really a good idea).
```js
await vs.hydrateNodeSecurePayload(payload.dependencies);
payload.vulnerabilityStrategy = vs.type;
```

And you need to update index.d.ts to add `vulnerabilityStrategy` to the **Payload** interface. You can create a separate type with value `"db_npm" | "db_security_wg"`.

Reply:
Alright that's true, I got it 😉